packages/expo-cli/src/commands/apply/configureIOSProjectAsync.ts

```diff
@@ -0,0 +1,90 @@
+import { IosPlist } from '@expo/xdl';
+import { IOSConfig, getConfig } from '@expo/config';
+import path from 'path';
+
+// TODO: come up with a better solution for using app.json expo.name in various places
+function sanitizedName(name: string) {
+  return name
+    .replace(/[\W_]+/g, '')
+    .normalize('NFD')
+    .replace(/[\u0300-\u036f]/g, '');
+}
+
+// TODO: it's silly and kind of fragile that we look at app config to determine the
+// ios project paths
+// * Overall this function needs to be revamped, just a placeholder for now!
+function getIOSPaths(projectRoot: string) {
+  const { exp } = getConfig(projectRoot, { skipSDKVersionRequirement: true });
+
+  let projectName = exp.name;
+  if (!projectName) {
+    throw new Error('Need a name ;O');
+  }
+
+  const iosProjectDirectory = path.join(projectRoot, 'ios', sanitizedName(projectName));
+  const iconPath = path.join(iosProjectDirectory, 'Assets.xcassets', 'AppIcon.appiconset');
+
+  return {
+    projectName,
+    iosProjectDirectory,
+    iconPath,
+  };
+}
+
+async function modifyEntitlementsPlistAsync(projectRoot: string, callback: (plist: any) => any) {
+  let entitlementsPath = IOSConfig.Entitlements.getEntitlementsPath(projectRoot);
+  let directory = path.dirname(entitlementsPath);
+  let filename = path.basename(entitlementsPath, 'plist');
+  await IosPlist.modifyAsync(directory, filename, callback);
+  await IosPlist.cleanBackupAsync(directory, filename, false);
+}
+
+async function modifyInfoPlistAsync(projectRoot: string, callback: (plist: any) => any) {
+  const { iosProjectDirectory } = getIOSPaths(projectRoot);
+  await IosPlist.modifyAsync(iosProjectDirectory, 'Info', callback);
+  await IosPlist.cleanBackupAsync(iosProjectDirectory, 'Info', false);
+}
+
+export default async function configureIOSProjectAsync(projectRoot: string) {
+  const { exp } = getConfig(projectRoot, { skipSDKVersionRequirement: true });
+
+  IOSConfig.BundleIdenitifer.setBundleIdentifierForPbxproj(projectRoot, exp.ios!.bundleIdentifier!);
```

Comment:
Probably should validate and throw an error here about the `bundleIdentifier`

Reply:
validate that we have a bundle identifier or what?